### PR TITLE
On demand activation

### PR DESCRIPTION
**Pull request by radicaled**
dart-tools will boot up analysis server and attached GUI components on demand if a Dart file is opened but there is no `.packages` or `pubspec.yaml` path present.

Work on #23.

### CODE REVIEW

**Comment by radicaled** [comment]
@devoncarew this PR makes some changes largeish to `analysis_server.coffee` and `dart-tools.coffee`, so beware if you're planning on touching those files.

File: lib/analysis_server.coffee [hunk 1]
```diff
@@ -108,6 +81,7 @@ class AnalysisServer
       callback(obj.event, obj)
 
   setAnalysisRoots: (paths) =>
+    @currentAnalysisRoots = new Set(paths)
```
Comment (devoncarew) [inline lib/analysis_server.coffee]: Perhaps log whenever we set the analysis roots? I'm not positive, but I suspect that this is an expensive operation.

**Comment by devoncarew** [comment]
> some changes largeish to analysis_server.coffee and dart-tools.coffee

Thanks for the heads up! I have some small changes to `dart-tools.coffee` - they'll survive the merge.

File: lib/analysis_server.coffee (hunk 1 above)
Comment (radicaled) [inline lib/analysis_server.coffee]: I did some unofficial performance tests, and it's actually pretty snappy. Not sure if it's just me (8GB 2011 MBP) or something else.

**Pull request merged**